Controllers/MoviesController.cs

```diff
@@ -0,0 +1,72 @@
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+using Movies.Api.Models;
+using Movies.Api.Repositories;
+
+namespace Movies.Api.Controllers;
+
+[ApiController]
+[Route("api/[controller]")]
+public class MoviesController : ControllerBase
+{
+    private readonly IMovieRepository _movieRepository;
```

Comment:
Repository pattern is a good pattern you will find it used in Project Core. Most of the recent service we used Vertical Slices with a bit different structure but it doesn't matter I explain later the differences and benefits.